lisa/features/nvme.py

```diff
@@ -153,10 +153,9 @@ def _remove_nvme_os_disk(self, disk_list: List[str]) -> List[str]:
         ):
             os_disk_nvme_device = self._get_os_disk_nvme_device()
             # Removing OS disk/device from the list.
-            for disk in disk_list:
+            for disk in disk_list.copy():
```

Comment:
The original logic is to remove one, the new logic is to remove all. Is there any reason to change it?

Reply:
Earlier logic only excludes only OS disk we need to exclude all remote disks not just OS disk. If same VM used for both premium 4k perf and resource disk perf/ nvme perf all remove disks also considered NVMe devices. Thats the reason we need to exclude all remote disks.

Reply:
Added extra details in comments as suggested.